src/pulse3D/plate_recording.py

```diff
@@ -600,7 +604,7 @@ def to_dataframe(self) -> pd.DataFrame:
                     )
 
         df = pd.DataFrame(data)
-        if not output_stim_data:
+        if output_stim_data and include_stim_data:
```

Comment:
What we can actually do here is just check if there's any stim data in the DF at all